infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -102,20 +102,30 @@ private boolean processMsg(LogReplicationEntryMsg txMessage) {
 
             for (OpaqueEntry opaqueEntry : newOpaqueEntryList) {
                 for (UUID streamId : opaqueEntry.getEntries().keySet()) {
-                    if (!streamMap.containsKey(streamId)) {
+                    if (ignoreEntriesForStream(streamId)) {
                         log.warn("Skip applying log entries for stream {} as it is noisy. LR could be undergoing a rolling upgrade", streamId);
                         continue;
                     }
 
                     List<SMREntry> smrEntries = opaqueEntry.getEntries().get(streamId);
                     if (streamId.equals(REGISTRY_TABLE_ID)) {
                         // Only retain tables that are not present in the registry table
-                        smrEntries = fetchNewEntries(new ArrayList<>(smrEntries));
-                    }
-
-                    for (SMREntry smrEntry : smrEntries) {
-                        // If stream tags exist for the current stream, it means its intended for streaming on the Sink (receiver)
-                        txnContext.logUpdate(streamId, smrEntry, dataStreamToTagsMap.get(streamId));
+                        List<SMREntry> newEntries = fetchNewEntries(new ArrayList<>(smrEntries));
+                        for (SMREntry smrEntry : newEntries) {
+                            // If stream tags exist for the current stream, it means its intended for streaming on the Sink (receiver)
+                            txnContext.logUpdate(streamId, smrEntry, config.getDataStreamToTagsMap().get(streamId));
+                        }
+                        if (!newEntries.isEmpty() && config.getStreamsDiscoveryMode().equals(StreamsDiscoveryMode.DYNAMIC)) {
+                            // In DYNAMIC mode, if there are new entries for registry table, update the config after
+                            // applying these new entries
+                            log.info("{} new entries for registry table found during log entry sync.", newEntries.size());
+                            config.syncWithRegistry();
```

Comment:
here too, the config can just be updated with the new entries instead of fully syncing with the registry table